### DISCUSSION TITLE
'active_scan' parameter has no effect on the function behaviour

### DISCUSSION THREAD

**Discussion by ahmedmoheb-nordic** [zephyrproject-rtos/zephyr#51210] [RFC]
Looking at this function
https://github.com/zephyrproject-rtos/zephyr/blob/e027f0671a20fe3ff3b315418d5a652ea277c1b5/subsys/bluetooth/host/id.c#L1527

'active_scan' parameter is used once in a logical OR operation with a `true` condition, as `!is_adv_using_rand_addr()` should have been evaluated to true at line 1550

Also that makes the following debug message unreachable 
https://github.com/zephyrproject-rtos/zephyr/blob/e027f0671a20fe3ff3b315418d5a652ea277c1b5/subsys/bluetooth/host/id.c#L1557

**Comment by ahmedmoheb-nordic** [comment]
@alwa-nordic @jori-nordic could you check this?

**Reply by alwa-nordic** [reply]
I'm not sure about this. This seems to be intentional. According to the commit message, the state of the advertiser can change between the two tests.

https://github.com/zephyrproject-rtos/zephyr/commit/f332d51d60c3e2603fab5ea82b148df3a809ac5a

```
commit f332d51d60c3e2603fab5ea82b148df3a809ac5a
Author: Joakim Andersson <joakim.andersson@nordicsemi.no>
Date:   Tue Sep 15 12:47:58 2020 +0200

    Bluetooth: host: Ignore failure to set passive scanner address
    
    Ignore error when failing to set the passive scanner private address.
    This can happen because in between the time we checked if the
    advertiser was enabled and the time we execute the set random address
    command the state of the advertiser could have changed in the
    controller, and we only set the state once the command has completed
    in the controller.
    
    Fixes: #25672
```
